### PR TITLE
fix(docs): Map `@alpha` tags to "Legacy" alerts in API documentation

### PR DESCRIPTION
**Pull request by Josmithr**
In our repo, we are currently using the `@alpha` release tag to mean something different than "alpha". We currently map it to a concept we're calling "legacy". This PR updates the generated API docs to use "Legacy" terminology in various tables in the API docs, instead of "Alpha" to avoid any customer confusion.

In the future, we plan on reclaiming `@alpha` for its traditional meaning, and using a `@legacy` tag for the special concept. The configurations changed here will be need to changed again at that time.

Before
image

After
image

[AB#8406](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/8406)

### CODE REVIEW

**Comment by Josmithr** [comment]
Pending #21524 being merged and published. At which time the dependency can be updated in this PR, and the new needed functionality can be leveraged.

File: tools/api-markdown-documenter/src/utilities/ApiItemUtilities.ts [hunk 1]
```diff
@@ -465,3 +466,19 @@ export function getSafeFilenameForName(apiItemName: string): string {
...
+ * @remarks Useful when a shortened version of a code excerpt is wanted, and especially if the contents need to fit on a single line.
+ * @public
+ */
+export function getSingleLineExcerptText(excerpt: Excerpt): string {
+	// Regex replaces line breaks with spaces to ensure everything ends up on a single line.
+	let signatureExcerpt = excerpt.text.trim().replace(/\s+/g, " ");
+
+	if (signatureExcerpt.endsWith(";")) {
+		signatureExcerpt = signatureExcerpt.slice(0, signatureExcerpt.length - 1);
+	}
```
Comment (alexvy86) [inline tools/api-markdown-documenter/src/utilities/ApiItemUtilities.ts:481]: So this function should be used just for code excerpts that are single-statement? If we had something like

```typescript
const a = 1;
const b = 2;
```

The transformed version would be `const a = 1; const b = 2` which feels a bit weird with the semicolon in the middle but none at the end.

Which I don't mind, but maybe worth calling out in remarks.

**Review by alexvy86 — APPROVED** [review]
Looks good as far as I can tell. Small nit below.

File: tools/api-markdown-documenter/src/utilities/ApiItemUtilities.ts (hunk 1 above)
Comment (Josmithr) [reply tools/api-markdown-documenter/src/utilities/ApiItemUtilities.ts:481]: That's an interesting point. That policy probably doesn't belong in this helper, if it's going to be public. Consumers can still trim off the trailing semi if they want. Though I want to double check what scenarios that's even relevant for.

Comment (Josmithr) [reply tools/api-markdown-documenter/src/utilities/ApiItemUtilities.ts:481]: I updated the base PR to omit the `;` trimming from the public function. The 2 internal use-cases still want it (specifically for heading text and table entry text for various signature types. Not critical in those scenarios, but nice to have.

**Comment by Josmithr** [comment]
> 🔗 Found some broken links! 💔

The link checker appears to be broken. We must not be building the API docs before running it. I'll investigate separately.

**Pull request merged**